presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -1950,6 +1951,54 @@ private void testBucketedExecution(Session session)
                 "SELECT custkey, COUNT(*) FROM orders GROUP BY custkey");
     }
 
+    @Test
+    public void testBucketPruning()
+    {
+        Session session = getSession();
+        QueryRunner queryRunner = getQueryRunner();
+        queryRunner.execute("CREATE TABLE orders_bucketed WITH (bucket_count = 11, bucketed_by = ARRAY['orderkey']) AS " +
+                "SELECT * FROM orders");
+
+        try {
+            assertQuery(session, "SELECT count(*) FROM orders_bucketed WHERE orderkey = 100", "SELECT 1");
```

Comment:
@gparai It would be better to use SELECT * queries. count(*) only confirms that row counts are correct, but * would confirm that all the data is right.

Reply:
Yes, you are correct! I have modified the tests.